File: backend/src/main/java/com/microshop/backend/entity/Question.java
package com.microshop.backend.entity;

import jakarta.persistence.*;
import lombok.Getter;
import lombok.Setter;
import org.hibernate.annotations.Nationalized;

@Getter
@Setter
@Entity
public class Question {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "question_id", nullable = false)
    private Integer id;

    @ManyToOne(fetch = FetchType.LAZY, optional = false)
    @JoinColumn(name = "subject_id", nullable = false)
    private Subject subject;

    @Nationalized
    @Column(name = "content", nullable = false, length = 500)
    private String content;

    @Nationalized
    @Column(name = "option_a")
    private String optionA;

    @Nationalized
    @Column(name = "option_b")
    private String optionB;

    @Nationalized
    @Column(name = "option_c")
    private String optionC;

    @Nationalized
    @Column(name = "option_d")
    private String optionD;

    @Column(name = "correct_option", nullable = false)
    private Character correctOption;

}

File: mobile/lib/data/models/question.dart
class Question {
  final int id;
  final int subjectId;
  final String content;
  final String optionA;
  final String optionB;
  final String optionC;
  final String optionD;
  final String correctOption; // 'A'/'B'/'C'/'D'

  Question({
    required this.id,
    required this.subjectId,
    required this.content,
    required this.optionA,
    required this.optionB,
    required this.optionC,
    required this.optionD,
    required this.correctOption,
  });

  factory Question.fromJson(Map<String, dynamic> j) => Question(
    id: j['id'],
    subjectId: j['subjectId'],
    content: j['content'],
    optionA: j['optionA'],
    optionB: j['optionB'],
    optionC: j['optionC'],
    optionD: j['optionD'],
    correctOption: (j['correctOption'] ?? '').toString(),
  );

  Map<String, dynamic> toJson() => {
    'id': id,
    'subjectId': subjectId,
    'content': content,
    'optionA': optionA,
    'optionB': optionB,
    'optionC': optionC,
    'optionD': optionD,
    'correctOption': correctOption,
  };
}
